### DISCUSSION TITLE
[WindowsAppSDK + WASM] Trouble when building control libraries

### DISCUSSION THREAD

**Discussion by DierkDroth** [unoplatform/uno#10676] [Q&A]
I'm setting up this solution structure:
```
- Base                             (.NET 7.0 assembly)
- Gui.Shared                       (shared project)
- Gui.Wasm.Library                 (UNO WASM control library)
- Gui.Windows.Library              (WindowsAppSDK control library)

- App.WASM                         (WASM app)
	|-> Gui.Wasm.Library
	|-> Base
- App.Windows                      (WindowsAppSDK app, MSIX packaged)			
	|-> Gui.Windows.Library
	|-> Base
```
... and running into a couple of issues.

Details:
- I'm using a customized version of the [UNO Material theme](https://platform.uno/docs/articles/features/uno-material.html)
- I'm aware of [this article](https://platform.uno/docs/articles/guides/how-to-create-control-libraries.html) and already replaced all `ms-appx` use cases by:
-- migrating assets to `Embedded resources` and loading them by [Assembly.GetManifestResourceStream](https://learn.microsoft.com/en-us/dotnet/api/system.reflection.assembly.getmanifestresourcestream) + respective per-case logic
-- `XamlReader.Load(MaterialTheme-XAML-master-resource)` (requires trickery to get the system to know some types in the XAML + adding `Microsoft.UI.Xaml.Data.Bindable` e.g. to  `FromBoolToValueConverter`)

I'm facing 2 problems:
- `Source = new Uri("ms-appx:///Material/MaterialTheme.xaml");` still is required for WASM, I could not get the `XamlReader.Load` approach to work with UNO WASM. It only works with Windows. 
**However, `ms-appx`  does not work** if `MaterialTheme.xaml` is placed in `Gui.Wasm.Library` , I had to put it in `App.WASM` `(Build action = Page)`.
(in fact it's a link to the instance in `Gui.Windows.Library ` - which goes with `Build action = Embedded Resource` - see above)
Any idea why that is?
- I get these warnings when building the WASM head
```
3>C:\Users\dierk\Documents\Unusual\Unusual.Wasm\Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs(7825,275,7825,316): warning CS0436: The type 'GlobalStaticResources' in 'Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs' conflicts with the imported type 'GlobalStaticResources' in 'Unusual.Gui.Wasm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs'.
3>C:\Users\dierk\Documents\Unusual\Unusual.Wasm\Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs(7854,275,7854,316): warning CS0436: The type 'GlobalStaticResources' in 'Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs' conflicts with the imported type 'GlobalStaticResources' in 'Unusual.Gui.Wasm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs'.
3>C:\Users\dierk\Documents\Unusual\Unusual.Wasm\Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs(7883,275,7883,316): warning CS0436: The type 'GlobalStaticResources' in 'Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs' conflicts with the imported type 'GlobalStaticResources' in 'Unusual.Gui.Wasm, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'. Using the type defined in 'Uno.UI.SourceGenerators\Uno.UI.SourceGenerators.XamlGenerator.XamlCodeGenerator\MaterialTheme_59319d1718c26a075d333a1c8f3cfbc6.cs'.
```

How could those be resolved (apart from ignoring in project file)?

**Comment by jeromelaban** [comment]
Thanks for the report. This is caused by https://github.com/unoplatform/uno/issues/8750. The workaround is to avoid using the same default namespace in both assemblies.

**Reply by DierkDroth** [reply]
Thanks @jeromelaban. I'll give it a try

Would you have any feedback on the first problem as well?

**Reply by jeromelaban** [reply]
You'll need to [specify the assembly name](https://learn.microsoft.com/en-us/windows/uwp/app-resources/uri-schemes#authority-ms-appx-and-ms-appx-web).

**Reply by DierkDroth** [reply]
Thanks for your feedback @jeromelaban. 

However, I fail to add an assembly name to the URI at all. Please see attached repo and check out `SharedResourceDictionary.cs` for comments

What am I doing wrong?

[UriSample.zip](https://github.com/unoplatform/uno/files/10182330/UriSample.zip)

**Reply by DierkDroth** [reply]
@jeromelaban would you have some feedback on my sample repo above? It could well be that I don't understand the idea of referencing the assembly by its name...

Thanks in advance
